From e84f25c237397ee5d24299fce7d562a2448f9a2b Mon Sep 17 00:00:00 2001
From: Honore Tricot <honorex.tricot@intel.com>
Date: Tue, 15 May 2012 14:35:12 +0200
Subject: [PORT FROM MAIN] WIFI: Move calibration file to /factory

BZ: 36278

When phone is encrypted and MAC is not available in secured partition,
provisioning tool restart calibration at every boot due to encryption
management prevent to read in /data patition till password is entered.

Fix constis in moving calibration file to /factory/wifi.

Change-Id: Ifc4571a9705c404717f2ab9e01e797c64691fa59
Orig-Change-Id: I6ec2e97b4904bb2909a7809a3d5aa0c3a4347608
Signed-off-by: Honore Tricot <honorex.tricot@intel.com>
Reviewed-on: http://android.intel.com:8080/48852
Reviewed-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: Garcia, JeremieX <jeremiex.garcia@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/59011
Reviewed-by: Regairaz, Julie <julie.regairaz@intel.com>
Reviewed-on: http://android.intel.com:8080/77962
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wlan_prov/Android.mk          |    2 +-
 wlan_prov/wlan_provisioning.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/wlan_prov/Android.mk b/wlan_prov/Android.mk
index ed94446..3b1e09a 100644
--- a/wlan_prov/Android.mk
+++ b/wlan_prov/Android.mk
@@ -36,7 +36,7 @@ $(SYMLINKS): $(LOCAL_INSTALLED_MODULE) $(LOCAL_PATH)/Android.mk
 	@echo "Symlink: $@"
 	@mkdir -p $(dir $@)
 	@rm -rf $@
-	$(hide) ln -sf $(addprefix /data/misc/firmware/ti-connectivity/,$(NVS_FILE)) $@
+	$(hide) ln -sf $(addprefix /factory/wifi/,$(NVS_FILE)) $@
 
 ALL_DEFAULT_INSTALLED_MODULES += $(SYMLINKS)
 ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index 71422d3..1e11073 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -37,7 +37,7 @@
 const unsigned char NullMacAddr[MAC_ADDRESS_LEN] = { 0, 0, 0, 0, 0, 0 };
 
 #define WIFI_PATH "/data/misc/wifi"
-const char NVS_file_name[] = "/data/misc/firmware/ti-connectivity/wl1271-nvs.bin";
+const char NVS_file_name[] = "/factory/wifi/wl1271-nvs.bin";
 const char Default_NVS_file_name[] = "/system/etc/wifi/wl1271-nvs.bin";
 const char WLAN_SDIO_BUS_PATH[] = "/sys/bus/sdio/drivers/wl1271_sdio/";
 #define WLAN_DRV_SDIO_NAME "wl1271_sdio"
-- 
1.7.4.1

